core/src/banking_stage/thread_aware_account_locks.rs

```diff
@@ -0,0 +1,692 @@
+use {
+    solana_sdk::pubkey::Pubkey,
+    std::{
+        collections::{hash_map::Entry, HashMap},
+        fmt::{Debug, Display},
+        ops::{BitAndAssign, Sub},
+    },
+};
+
+pub const MAX_THREADS: usize = u64::BITS as usize;
+
+/// Identifier for a thread
+pub type ThreadId = usize; // 0..MAX_THREADS-1
+
+/// A bit-set of threads an account is scheduled or can be scheduled for.
+#[derive(Copy, Clone, PartialEq, Eq)]
+#[repr(transparent)]
+pub struct ThreadSet(u64);
+
+/// Thread-aware account locks which allows for scheduling on threads
+/// that already hold locks on the account. This is useful for allowing
+/// queued transactions to be scheduled on a thread while the transaction
+/// is still being executed on the thread, up to a queue limit.
+pub struct ThreadAwareAccountLocks {
+    /// Number of threads.
+    num_threads: usize, // 0..MAX_THREADS
+    /// Limit on the number of sequentially-queued transactions per account.
+    sequential_queue_limit: u32,
```

Comment:
nit: naming: `sequential_lock_limit` for fewer nouns. i think `queue` is a bit foreign in this context, coming from higher layer logics (i.e. actual scheduler code)?

Reply:
as discussed elsewhere, the sequential_lock_limit was entirely removed from this struct: 6a8a0a89fd20d3a91ec936828df6e0052dbf0fed